### PR TITLE
Change the available Benchmark options

### PR DESCRIPTION
**Pull request by EricWF**
This patch does the following:

* Move `State::UseRealTime()` to `Benchmark::UseRealTime()`.
* Add `Benchmark::MinTime(double t)` option to allow the min time to be set on a per-test basis.
* Remove `--benchmark_iterations=<n>` option. Only allow time based tests.

### CODE REVIEW

File: src/benchmark.cc [hunk 1]
```diff
@@ -101,6 +97,10 @@ GetBenchmarkLock()
 
 namespace {
 
+bool IsZero(double n) {
+    return std::abs(n) < std::numeric_limits<double>::epsilon();
```
Comment (dominichamon) [inline src/benchmark.cc]: std::fabs

File: src/benchmark.cc [hunk 2]
```diff
@@ -424,6 +428,12 @@ bool BenchmarkFamilies::FindBenchmarks(
         if (family->arg_count_ >= 2) {
           AppendHumanReadable(instance.arg2, &instance.name);
         }
+        if (!IsZero(family->min_time_)) {
+          instance.name +=  StringPrintF("/min_time:%0.3f",  family->min_time_);
```
Comment (dominichamon) [inline src/benchmark.cc]: you'll need to take this into account in the width of the first column in tabular format.

File: src/benchmark.cc [hunk 3]
```diff
@@ -493,6 +504,15 @@ void BenchmarkImp::RangePair(int lo1, int hi1, int lo2, int hi2) {
   }
 }
 
+void BenchmarkImp::MinTime(double t) {
+  CHECK(t > 0.0);
```
Comment (dominichamon) [inline src/benchmark.cc]: what if i set MinTime(0.0) to disable the minimum time for a given benchmark?

File: test/benchmark_test.cc [hunk 4]
```diff
@@ -70,7 +68,7 @@ static void BM_FactorialRealTime(benchmark::State& state) {
   ss << fac_42;
   state.SetLabel(ss.str());
 }
-BENCHMARK(BM_FactorialRealTime);
+BENCHMARK(BM_FactorialRealTime)->UseRealTime();
```
Comment (dominichamon) [inline test/benchmark_test.cc]: no need for this .. just add this macro to the BM_Factorial benchmark above to show how it works.

File: src/benchmark.cc (hunk 1 above)
Comment (EricWF) [inline src/benchmark.cc]: Why? I'm not opposed but std::abs(...) has overloads for floating point types.

Comment (dominichamon) [inline src/benchmark.cc]: oh that new standard library! It didn't used to.

File: src/benchmark.cc (hunk 3 above)
Comment (EricWF) [inline src/benchmark.cc]: How long does the benchmark run for then? The maximum number of iterations? I think setting it to 0.0 is likely a bug and not a feature.

File: src/benchmark.cc (hunk 2 above)
Comment (EricWF) [inline src/benchmark.cc]: The way we calculate the width seems weird. Shouldn't it just be the size of the longest name plus room for `_stddev`?

File: test/benchmark_test.cc (hunk 4 above)
Comment (EricWF) [inline test/benchmark_test.cc]: ack.

File: src/benchmark.cc (hunk 2 above)
Comment (EricWF) [inline src/benchmark.cc]: I'll fix the way the column width is calculated.

Comment (dominichamon) [inline src/benchmark.cc]: i didn't want to mess with it too much because it was doing voodoo.

**Pull request merged**